compiler-plugin/src/main/kotlin/com/google/devtools/ksp/processing/impl/ResolverImpl.kt

```diff
@@ -1363,13 +1362,11 @@ class ResolverImpl(
             RefPosition.PARAMETER_TYPE -> typeSystem.getOptimalModeForValueParameter(kotlinType)
             RefPosition.RETURN_TYPE ->
                 typeSystem.getOptimalModeForReturnType(kotlinType, ref.isReturnTypeOfAnnotationMethod())
+
             RefPosition.SUPER_TYPE -> TypeMappingMode.SUPER_TYPE
         }.updateFromParents(ref)
 
-        val parameters = kotlinType.constructor.parameters
-        val arguments = kotlinType.arguments
-
-        parameters.zip(arguments).forEach { (parameter, argument) ->
```

Comment:
This has a very very slight change where if the parameters and arguments had different size this would use the smaller of the sizes but they should always have been the same size.